src/wp-includes/block-supports/block-style-variations.php

```diff
@@ -422,3 +407,93 @@ function wp_enqueue_block_style_variation_styles() {
 add_filter( 'wp_theme_json_data_theme', 'wp_resolve_block_style_variations_from_styles_registry', 10, 1 );
 
 add_filter( 'wp_theme_json_data_user', 'wp_resolve_block_style_variations_from_theme_style_variation', 10, 1 );
+
+/**
+ * @access private
+ */
+function wp_register_block_style_variations_from_theme_json_data( $variations ) {
+	$variations_data = array();
+
+	if ( empty( $variations ) ) {
+		return $variations_data;
+	}
```

Comment:
I don't think we need this `$variations_data` variable now as we aren't filtering theme.json data here.